### PR TITLE
update pyproject readme docs

### PR DESCRIPTION
**Pull request by patrick-ryan**
# Pull Request Check List

Resolves: #5069

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

I rendered the file in vscode, but let me know if there's a way to generate + host the docs locally so I can preview, thanks!

### CODE REVIEW

File: docs/pyproject.md [hunk 1]
```diff
@@ -69,9 +69,21 @@ This is a list of maintainers and should be distinct from authors. Maintainers m
...
+`pyproject.toml`.
+
+The file contents becomes the
+[Description field](https://packaging.python.org/en/latest/specifications/core-metadata/#description-optional)
+in a PyPI build (similar to `long_description` in setuptools).
+
+```toml
+[tool.poetry]
+# ...
+readme = "./README.md"
```
Comment (Secrus) [inline docs/pyproject.md:85]: I don't think we need `./` prefix. It is unnecessary, as it points to the same directory and can be confusing.

**Review by Secrus — CHANGES_REQUESTED** [review]

File: docs/pyproject.md (hunk 1 above)
Comment (patrick-ryan) [reply docs/pyproject.md]: okay np I fixed that, looks better now

**Review by Secrus — DISMISSED** [review]

**Comment by neersighted** [comment]
Yeah, we've added support for `readme` being a list:

https://github.com/python-poetry/poetry-core/blob/b7c99733539aa443313c0d08fd4b85f8e5230bfc/src/poetry/core/factory.py#L146

A quick touchup to reflect that would be appreciated!

**Comment by Secrus** [comment]
@patrick-ryan are you still up for completing this?

**Comment by patrick-ryan** [comment]
yep I'll take a look at this, thanks for the ping

**Comment by patrick-ryan** [comment]
okay updated to include the option of specifying a list of readme paths, let me know if that looks okay

File: docs/pyproject.md [hunk 2]
```diff
@@ -106,14 +106,28 @@ maintainers = [
 
 ## readme
 
-The readme file of the package. **Optional**
+The path, or array of paths, to the readme file(s) of the package. **Optional**
 
-The file can be either `README.rst` or `README.md`.
+The file should be in either Markdown (.md) or reStructuredText (.rst)
```
Comment (Secrus) [inline docs/pyproject.md:111]: As #5357 is already present, this is no longer true. Somehow adding this to docs here was missed. Could you add this, please?

**Review by Secrus — CHANGES_REQUESTED** [review]

File: docs/pyproject.md (hunk 2 above)
Comment (patrick-ryan) [reply docs/pyproject.md:111]: oh nice, yes I will add that

Comment (patrick-ryan) [reply docs/pyproject.md]: okay that's added now

File: docs/pyproject.md [hunk 3]
```diff
@@ -106,12 +106,27 @@ maintainers = [
 
 ## readme
 
-The readme file of the package. **Optional**
+The path, or array of paths, to the readme file(s) of the package. **Optional**
```
Comment (neersighted) [inline docs/pyproject.md:109]:
```suggestion
A path, or list of paths corresponding to the README file(s) of the package. **Optional**
```

File: docs/pyproject.md [hunk 4]
```diff
@@ -106,12 +106,27 @@ maintainers = [
...
 
-The file can be either `README.rst` or `README.md`.
+The file can be any format, but if you intend to publish to PyPI keep the
```
Comment (neersighted) [inline docs/pyproject.md:111]:
```suggestion
The file(s) can be of any format, but if you intend to publish to PyPI keep the
```

File: docs/pyproject.md [hunk 5]
```diff
@@ -106,12 +106,27 @@ maintainers = [
...
+[recommendations for a PyPI-friendly README](
+https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/)
+in mind. The path is relative to the directory containing the `pyproject.toml`.
```
Comment (neersighted) [inline docs/pyproject.md:114]:
```suggestion
in mind. README paths are implicitly relative to `pyproject.toml`.
```

File: docs/pyproject.md [hunk 6]
```diff
@@ -106,12 +106,27 @@ maintainers = [
...
+
+The file contents becomes the
```
Comment (neersighted) [inline docs/pyproject.md:116]:
```suggestion
The contents of the README file(s) are used to populate the
```

**Review by neersighted — CHANGES_REQUESTED** [review]

File: docs/pyproject.md [hunk 7]
```diff
@@ -106,12 +106,27 @@ maintainers = [
...
 
-The file can be either `README.rst` or `README.md`.
+The file(s) can be of any format, but if you intend to publish to PyPI keep the
+[recommendations for a PyPI-friendly README](
+https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/)
+in mind. README paths are implicitly relative to `pyproject.toml`.
+
+The contents of the README file(s) are used to populate the
+[Description field](https://packaging.python.org/en/latest/specifications/core-metadata/#description-optional)
+of your distribution's metadata (similar to `long_description` in setuptools). When specifying multiple files are specified, they are concatenated with newlines.
```
Comment (neersighted) [inline docs/pyproject.md:118]:
```suggestion
of your distribution's metadata (similar to `long_description` in setuptools). When multiple files are specified they are concatenated with newlines.
```

Comment (neersighted) [reply docs/pyproject.md:118]: My bad on the suggestions here :)

**Review by neersighted — CHANGES_REQUESTED** [review]

File: docs/pyproject.md (hunk 7 above)
Comment (patrick-ryan) [reply docs/pyproject.md]: np I had fixed this locally, it's now updated along with column wrap

**Review by neersighted — APPROVED** [review]

**Pull request merged**